Makefile

```
@@ -32,6 +32,50 @@ else ifeq ($(OS_GENERAL),Windows)
 	OS_PACKAGE_MANAGER := choco
 endif
 
+# Detect the operating system
+UNAME_S := $(shell uname -s)
+
+# Directory where binaries will be placed
+BIN_DIR := ./libs
+
+# Set the library file name based on the operating system
+ifeq ($(UNAME_S),Linux)
```

todo: this and all others.

```suggestion
ifeq ($(OS_GENERAL),Linux)
```

---

changed